### PR TITLE
initial commit of Test-DbaFullRecoveryModel

### PR DESCRIPTION
**Pull request by ClaudioESSilva**
Ready for QA

### CODE REVIEW

**Comment by ClaudioESSilva** [comment]
Set of tests for this PR
SQL Server 2000

- [ ] Don't run on version 2000 and throw an message

SQL Server 2005 - 2016 (perform by this order **without -Detailed switch**)
- [ ] Create Database with SIMPLE recovery model. Run the command. The database does not show up
- [ ] Change Database to BULK LOGGED recovery model. Run the command. The database does not show up
- [ ] Change Database to FULL recovery model. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'pseudo-SIMPLE'
- [ ] Take a full backup on database. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'FULL'
- [ ] Change Database to SIMPLE recovery model. Then change back to FULL recovery model. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'pseudo-SIMPLE'
- [ ] Take a full backup on database. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'FULL'

SQL Server 2005 - 2016 (perform by this order **WITH -Detailed switch**)
- [ ] Create Database with SIMPLE recovery model. Run the command. The database does not show up
- [ ] Change Database to BULK LOGGED recovery model. Run the command. The database does not show up
- [ ] Change Database to FULL recovery model. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'pseudo-SIMPLE' and 'Notes' column must say "Database is still in SIMPLE recovery model until a full database backup is taken."
- [ ] Take a full backup on database. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'FULL' and 'Notes' column must be empty
- [ ] Change Database to SIMPLE recovery model. Then change back to FULL recovery model. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'pseudo-SIMPLE'  and 'Notes' column must say "Database is still in SIMPLE recovery model until a full database backup is taken."
- [ ] Take a full backup on database. Run the command. The database does show up and 'ActualRecoveryModel' column must say 'FULL' and 'Notes' column must be empty

File: functions/Test-DbaFullRecoveryModel.ps1 [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+Only shows the databases that are in 'pseudo-simple' mode.
+	
+.EXAMPLE   
+Test-DbaFullRecoveryModel -SqlServer sql2008 | Sort-Object Server, ActualRecoveryModel -Descending
+	
+Shows all databases which actual configured recovery model is FULL and says if they are really in FULL recovery model or not. Will show in first place the ones that are in 'pseudo-simple' mode.
+	
+#>
+	[CmdletBinding(SupportsShouldProcess = $true)]
```
Comment (dansqldba) [inline functions/Test-DbaFullRecoveryModel.ps1]: Not needed for a test cmdlet?

File: functions/Test-DbaFullRecoveryModel.ps1 [hunk 2]
```diff
@@ -0,0 +1,169 @@
+﻿Function Test-DbaFullRecoveryModel
+{
+<#
+.SYNOPSIS
+Find if database is really in the Full recovery model or not
+
+.DESCRIPTION
+When you switch a database into FULL recovery model, it will behave like a SIMPLE recovery model until a full backup is taken in order to begin a log backup chain.
+This state is alson known as 'pseudo-Simple'.
```
Comment (dansqldba) [inline functions/Test-DbaFullRecoveryModel.ps1]: Typo: "alson"

File: functions/Test-DbaFullRecoveryModel.ps1 [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	{
+        foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $SqlCredential
+
+                if ($server.versionMajor -lt 9)
+		        {
+			        Write-Warning "This function does not support versions lower than SQL Server 2005 (v9)"
```
Comment (dansqldba) [inline functions/Test-DbaFullRecoveryModel.ps1]: Which instance? Please include the instance name in the output.

File: functions/Test-DbaFullRecoveryModel.ps1 (hunk 2 above)
Comment (ClaudioESSilva) [inline functions/Test-DbaFullRecoveryModel.ps1]: Done

File: functions/Test-DbaFullRecoveryModel.ps1 (hunk 1 above)
Comment (ClaudioESSilva) [inline functions/Test-DbaFullRecoveryModel.ps1]: Done

**Comment by dansqldba** [comment]
Not sure how we feel about Write-Host? I've let it slide to get this cmdlet into the module ;)

**Pull request merged**